src/ripple/core/SQLInterface.h

```diff
@@ -0,0 +1,224 @@
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CORE_SQLINTERFACE_H_INCLUDED
+#define RIPPLE_CORE_SQLINTERFACE_H_INCLUDED
+
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/app/main/Application.h>
+#include <ripple/app/misc/Transaction.h>
+#include <ripple/core/Config.h>
+#include <ripple/core/DatabaseCon.h>
+#include <ripple/peerfinder/impl/Store.h>
+#include <boost/filesystem.hpp>
+#include <boost/optional.hpp>
+#include <boost/variant.hpp>
+
+namespace ripple {
+
+struct LedgerHashPair
+{
+    uint256 ledgerHash;
+    uint256 parentHash;
+};
+
+class SQLInterface
+{
+public:
+    struct CountMinMax
+    {
+        int numberOfRows;
+        int minLedgerSequence;
+        int maxLedgerSequence;
+    };
+
+    struct AccountTxMarker
+    {
+        uint32_t ledgerSeq = 0;
+        uint32_t txnSeq = 0;
+    };
+
+    struct AccountTxOptions
+    {
+        AccountID const& account;
+        int minLedger;
+        int maxLedger;
+        std::uint32_t offset;
+        int limit;
+        bool bUnlimited;
+    };
+
+    struct AccountTxPageOptions
+    {
+        AccountID const& account;
+        int minLedger;
+        int maxLedger;
+        std::optional<AccountTxMarker> marker;
+        int limit;
+        bool bAdmin;
+    };
+
+    using AccountTx =
+        std::pair<std::shared_ptr<Transaction>, std::shared_ptr<TxMeta>>;
+    using AccountTxs = std::vector<AccountTx>;
+    using txnMetaLedgerType = std::tuple<Blob, Blob, std::uint32_t>;
+    using MetaTxsList = std::vector<txnMetaLedgerType>;
+
+    static std::unique_ptr<SQLInterface>
+    init(Application& app, Config const& config, JobQueue& jobQueue);
+
+    virtual ~SQLInterface() = default;
+
+    virtual std::optional<LedgerIndex>
+    getMinLedgerSeq() = 0;
+
+    virtual std::optional<LedgerIndex>
+    getTransactionsMinLedgerSeq() = 0;
+
+    virtual std::optional<LedgerIndex>
+    getAccountTransactionsMinLedgerSeq() = 0;
+
+    virtual std::optional<LedgerIndex>
+    getMaxLedgerSeq() = 0;
+
+    virtual void
+    deleteTransactionByLedgerSeq(LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteBeforeLedgerSeq(LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteTransactionsBeforeLedgerSeq(LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteAccountTransactionsBeforeLedgerSeq(LedgerIndex ledgerSeq) = 0;
+
+    virtual int
+    getTransactionCount() = 0;
+
+    virtual int
+    getAccountTransactionCount() = 0;
+
+    virtual struct CountMinMax
+    getLedgerCountMinMax() = 0;
+
+    virtual bool
+    saveValidatedLedger(
+        std::shared_ptr<Ledger const> const& ledger,
+        bool current) = 0;
+
+    virtual std::optional<LedgerInfo>
+    loadLedgerInfoByIndex(LedgerIndex ledgerSeq) = 0;
```

Comment:
The interface seems to use `load` and `get` interchangeably. For example `loadTxHistory()` versus `getNewestAccountTxs()`. I think we should use the same verb for all of these functions, and I think `get` is better. So `getTxHistory()`, `getNewestLedgerInfo()`, etc.

Reply:
Done.